tracee-ebpf/tracee/tracee.go

```diff
@@ -1108,6 +1151,35 @@ func (t *Tracee) processEvent(ctx *context, args map[argTag]interface{}) error {
 	return nil
 }
 
+func getFileHash(fileName string) string {
+	f, _ := os.Open(fileName)
+	if f != nil {
+		defer f.Close()
+		h := sha256.New()
+		_, _ = io.Copy(h, f)
+		return hex.EncodeToString(h.Sum(nil))
+	}
+	return ""
+}
+
+func (t *Tracee) updateProfile(ctx *context, sourceFilePath string, sourceFileCtime int64) {
+	sourceFileSHA := getFileHash(sourceFilePath)
+	if pf, ok := t.profiledFiles[sourceFilePath]; !ok {
```

Comment:
sourceFilePath is not a good key for this map, as files may be overwritten. I think a decent key can be capturedFileID + sourceFileCtime